### ISSUE TITLE
tsx does not have any display

### ISSUE THREAD

**Issue by l-zoy** [vuejs/eslint-plugin-vue#1454]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:7.21.0 
- **eslint-plugin-vue version:7.7.0
- **Node version:15.8.0 
- **Operating System:Mac os 11.2.2

**Please show your full configuration:**

```json5
{
  env: { browser: true, es6: true, node: true, jest: true },
  globals: {},
  noInlineConfig: undefined,
  parser: 'vue-eslint-parser',
  parserOptions: {
    extraFileExtensions: [ '.vue' ],
    ecmaFeatures: {
      jsx: true,
    },
    // jsx: true tried ,
    ecmaVersion: 2020,
    sourceType: 'module',
    parser: '@typescript-eslint/parser'
  },
  plugins: [ 'import', 'prettier', '@typescript-eslint', 'vue' ],
}
```

**What did you do?**	

```vue
import { defineComponent } from 'vue'

const a = 1

export default defineComponent({
  name: 'tsx',
  data() {
    return {
      tsx: 'hello vue tsx'
    }
  },

  render() {
    return <div>123</div>
  }
})

```

**What did you expect to happen?**
report "a" unused 

**What actually happened?**

Nothing actually happened, the editor eslint output did not report an error, and if use the .jsx file between, the error is reported normally

**Repository to reproduce this issue**

**Comment by l-zoy** [comment]
It has been resolved and found that @vue/eslint-config-typescript originally needed plugin:vue/vue3-recommended to be put in front😅
